### PR TITLE
feat: Native asset send

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## **Description**

Implements redesigned native asset transfer (Simple send) both for wallet initiated and dApp initiated confirmations. Includes e2e tests.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27979?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3028 https://github.com/MetaMask/MetaMask-planning/issues/3027

## **Manual testing steps**

1. Go to this page...
2.
3.

### **After**

Screenshot 2024-10-21 at 10 32 05

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by jpuri — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/info/info.tsx [hunk 1]
```diff
@@ -34,6 +35,7 @@ const Info = () => {
       [TransactionType.tokenMethodTransfer]: () => TokenTransferInfo,
       [TransactionType.tokenMethodTransferFrom]: () => NFTTokenTransferInfo,
       [TransactionType.tokenMethodSafeTransferFrom]: () => NFTTokenTransferInfo,
+      [TransactionType.simpleSend]: () => NativeTransferInfo,
```
Comment (digiwand) [inline ui/pages/confirmations/components/confirm/info/info.tsx:38]: [nit] relocate to under `TransactionType.deployContract` to aim for alphabetizing. We should alphabetize the other items in this list as it is growing

**Review by digiwand — CHANGES_REQUESTED** [review]
hi @pedronfigueiredo 👋🏼 

there is a precision issue here. We are able to add lengthy values in the Send screen e.g. `0.000123456789012345678901234567890123456789` (paste twice in initial input). Because JavaScript coerces numbers outside of the MAX_SAFE_INTEGER range into scientific notation, it loses precision. 

From the Send screen, we can observe that the From value can have our desired number. Then we can see the To value loses decimal values. In the Review screen the value is also seen to have lost decimals from the original input. 

I'm unsure what the expected behavior here is, but if we did want to allow the users to enter in more digits, we could support this by preserving the string value and passing this string value to the BigNumber constructor.

once that is done, we'd need to be careful to avoid coercing it back into a JS number because it would lose precision again. We could use toFixed or toString to display the values, and avoid toNumber here

example BigNumber handling:
```
  const transferValue = new BigNumber(
    '0.000123456789012345678901234567890123456789',
  );

  console.log('transferValue', transferValue);
  console.log('transferValue toNumber', transferValue?.toNumber());
  console.log('transferValue toFixed', transferValue?.toFixed());
  console.log('transferValue toString', transferValue?.toString());
```
CleanShot 2024-10-29 at 16 00 36@2x

observe large value displayed input:


observe value has been coerced in To value


observe value has been coerced in Review screen

**Comment by pedronfigueiredo** [comment]
@digiwand In practice, I don't think there is any issue.

The reason you're seeing a truncated value is that the transfer value is saved as a WEI value.

You can see that on `nativeSendHeading`, `nativeAssetTransferValue` converts that value back to ETH.

I think it's reasonable not to expect resolutions below WEI units. In other words, the user is limited by the number of decimals the token has. What I don't know for sure is wether or not the number of decimals for native tokens can be determined dynamically or not with existing utils in the wallet, but I haven't found that. So at the moment that number is hardcoded to 18.

**Review by jpuri — DISMISSED** [review]

**Review by digiwand — DISMISSED** [review]

**Review by jpuri — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/info/info.tsx [hunk 2]
```diff
@@ -25,15 +29,13 @@ const Info = () => {
         }
         return TypedSignInfo;
       },
-      [TransactionType.contractInteraction]: () => BaseTransactionInfo,
-      [TransactionType.deployContract]: () => BaseTransactionInfo,
       [TransactionType.tokenMethodApprove]: () => ApproveInfo,
       [TransactionType.tokenMethodIncreaseAllowance]: () => ApproveInfo,
+      [TransactionType.tokenMethodSafeTransferFrom]: () => NFTTokenTransferInfo,
```
Comment (digiwand) [inline ui/pages/confirmations/components/confirm/info/info.tsx:34]: ❤️

**Review by digiwand — DISMISSED** [review]

**Review by jpuri — APPROVED** [review]

**Review by vinistevam — APPROVED** [review]

**Pull request merged**